src/leader.py

```diff
@@ -62,12 +62,17 @@ def turn_right(self):
     def turn_straight(self):
         self.fw.turn(self.STRAIGHT_ANGLE)
 
-
-    def turn(self,position):
-        if position == -1:
-            self.turn_left()
-        elif position == 1:
-            self.turn_right()
+    def set_turn_angle(self,position):
+        self.fw.turn(self.STRAIGHT_ANGLE + 45  * position)
+    
+    def turn(self,position,code):
+        if code == 'dpad_left_right':
+            if position == -1:
+                self.turn_left()
+            elif position == 1:
+                self.turn_right()
```

Comment:
You still need to check if position == 0 to straighten the wheels. As it is now, the wheels will never straighten. You only call this function if the code is dpad_left_right or left_stick_x, so it will never fall into this else block.